docs/source/place_and_route.rst

```diff
@@ -0,0 +1,317 @@
+Vertex
+    A *vertex* in *application graph*. Each vertex is mapped onto exactly one
+    SpiNNaker chip by during the placement process. (Note: an individual
+    SpiNNaker chip may have several *vertices* mapped to it). A vertex may
+    consume a certain set of *resources*. In most applications a vertex will
+    represent an application to be run on a single SpiNNaker core.
+    
+    *Vertices* are represented by application-defined :py:class:`object`\ s
+    which implement :py:meth:`object.__eq__` and :py:meth:`object.__hash__`.
+Net
+    A (directed) connection from one *vertex* to a number of other *vertices*
+    in the *application graph*. During routing, nets are converted into
+    specific routes through a SpiNNaker machine which can be used to generate
+    routing tables.
+    
+    *Nets* are represented by instances of the :py:class:`rig.netlist.Net`
+    class.
+Resource
+    A *resource* is any finite resource available to a SpiNNaker chip (e.g.
+    SDRAM) which may be consumed by a vertex. *Resources* are allocated to each
+    *vertex* during allocation. Users are welcome to define their own
+    application-specific resources.
+    
+    The type of a *resource* is represented by some unique python
+    :py:class:`object`. Some common resources are defined in
+    :py:mod:`rig.machine` (though users are free to use their own):
+    
+    * :py:data:`rig.machine.Cores`: Resource identifier for (monitor and
+      application) processor cores.
+    
+    * :py:data:`rig.machine.SDRAM`: Resource identifier for shared off-die
+      SDRAM (in bytes).
+    
+    * :py:data:`rig.machine.SRAM`: Resource identifier for shared on-die SRAM
+      (in bytes).
+    
+    Quantities of a *resource* are represented by positive integer values.
+Constraint
+    *Constraints* specify additional requirements on how an application graph is
+    placed and routed. For example a constraint might be used to force a
+    particular *vertex* to always be placed on a specific chip.
+    
+    A number of types of *constraint* are defined in
+    :py:class:`rig.place_and_route.constraints`.
+
+.. note::
+    It is worth emphasising that vertices being placed on SpiNNaker *chips*,
+    not specific cores. In this library, cores are just one of many chip
+    resources which vertices may consume.
+    
+    For most applications, each vertex represents exactly one core worth of
+    work and so each vertex will consume a single core of spinnaker chip
+    resource.
+    
+    Vertices which consume no cores are typically only useful when describing
+    external devices connected to the SpiNNaker system.
+    
+    Vertices which consume more than one core are typically only useful when a
+    vertex represents a group of applications which share memory. This is
+    because vertices will always be placed on a single SpiNNaker chip: they
+    cannot be split accross many chips. If an application requires this type of
```

Comment:
Ta. Also, joy: vim's spellcheck doesn't check inside `.. note::` ughhh I dislike rst and the universe around it...